implementation/src/main/java/io/smallrye/config/Converters.java

```diff
@@ -562,26 +566,44 @@ public A convert(final String str) {
 
     static final class BuiltInConverter<T> implements Converter<T>, Serializable {
         private final int id;
+        private final StringCleanup stringCleanup;
         private final Converter<T> function;
 
         static <T> BuiltInConverter<T> of(int id, Converter<T> function) {
-            return new BuiltInConverter<>(id, function);
+            return new BuiltInConverter<>(id, StringCleanup.NONE, function);
+        }
+
+        static <T> BuiltInConverter<T> of(int id, StringCleanup stringCleanup, Converter<T> function) {
+            return new BuiltInConverter<>(id, stringCleanup, function);
         }
 
-        private BuiltInConverter(final int id, final Converter<T> function) {
+        private BuiltInConverter(final int id, final StringCleanup stringCleanup, final Converter<T> function) {
             this.id = id;
+            this.stringCleanup = stringCleanup;
             this.function = function;
         }
 
         public T convert(final String value) {
-            return function.convert(value);
+            final String cleanValue;
+            if (value != null && StringCleanup.TRIM == stringCleanup) {
+                cleanValue = value.trim();
+            } else {
+                cleanValue = value;
+            }
+
+            return function.convert(cleanValue);
         }
 
         Object writeReplace() {
             return new Ser(id);
         }
     }
 
+    enum StringCleanup {
```

Comment:
I can see that using an Enum vs a Boolean means we can easily expand the behavior in the future.

But I wonder whether we would need anything here beyond "trim" and "don't trim". Can you think of any string manipulation other than trim we might need in the future? If not, I'd lean towards just using a boolean

Reply:
It might even make more sense to wrap the converter with a `TrimmingConverter` which could be reused.

Reply:
That's a cool idea, then it can be easily added/removed from converters by us, and used separately by others if desired

Reply:
It overlaps with another change I was doing so I'll send a single PR which supercedes this one.  /cc @yrodiere

Reply:
Opened #185.  I borrowed a commit from this PR.